storage/src/vespa/storage/common/global_bucket_space_distribution_converter.cpp

```diff
@@ -155,14 +152,14 @@ void build_global_groups(DistributionConfigBuilder& builder, const DistributionC
 std::shared_ptr<DistributionConfig>
 GlobalBucketSpaceDistributionConverter::convert_to_global(const DistributionConfig& source) {
     DistributionConfigBuilder builder;
-    set_distribution_invariant_config_fields(builder, source);
+    set_distribution_invariant_config_fields(builder);
     build_global_groups(builder, source);
     return std::make_shared<DistributionConfig>(builder);
 }
 
 std::shared_ptr<lib::Distribution>
 GlobalBucketSpaceDistributionConverter::convert_to_global(const lib::Distribution& distr) {
-    const auto src_config = distr.serialize();
+    const auto & src_config = distr.serialize();
```

Comment:
If the returned value of `serialize()` stops being a reference (an implementation detail), this will depend on implicit lifetime extension to work as expected, which I personally tend to find a bit too magic for my liking. This is not a hot path, so consider not taking by ref.

Reply:
Fixed